lib/cocoapods/installer/xcode/pods_project_generator/project_generator.rb

```diff
@@ -49,35 +54,36 @@ class ProjectGenerator
         # @param [String] podfile_path @see #podfile_path
         #
         def initialize(sandbox, path, pod_targets, build_configurations, platforms,
-                       object_version = nil, podfile_path = nil)
+                       object_version = nil, podfile_path = nil, pod_target_subproject: false)
           @sandbox = sandbox
           @path = path
           @pod_targets = pod_targets
           @build_configurations = build_configurations
           @platforms = platforms
           @object_version = object_version
           @podfile_path = podfile_path
+          @pod_target_subproject = pod_target_subproject
         end
 
         public
 
         # @return [Project] Generated and prepared project.
         #
         def generate!
-          project = create_project(path, object_version)
+          project = create_project(path, object_version, pod_target_subproject)
           prepare(sandbox, project, pod_targets, build_configurations, platforms, podfile_path)
           project
         end
 
         private
 
-        def create_project(path, object_version)
+        def create_project(path, object_version, pod_target_subproject)
           object_version ||= Xcodeproj::Constants::DEFAULT_OBJECT_VERSION
-          Pod::Project.new(path, false, object_version)
+          Pod::Project.new(path, false, object_version, :pod_target_subproject => pod_target_subproject)
         end
 
         def prepare(sandbox, project, pod_targets, build_configurations, platforms, podfile_path)
-          UI.message '- Creating Pods project' do
+          UI.message "- Creating #{project.path.basename('.*')} project" do
```

Comment:
maybe it makes sense to add a method to `CocoaPods::Project` that returns the project basename?

Reply:
yeah same comment/question as I had above, can we consolidate all of these `.basename('.*')` into a single method

Reply:
Created a new property `project_name`